R/data-set.R

```diff
@@ -0,0 +1,141 @@
+#' @title DataSet
+#' @docType class
+#' @description  A wrapper around DataRepository exposing convenience methods to use and manipulate dataSets
+#' (typically observed data) containing an X column, a Y column and potentially an Error columns
+#' @format NULL
+DataSet <- R6::R6Class(
+  "DataSet",
+  inherit = Printable,
+  cloneable = FALSE,
+  active = list(
+    #' @field name The name of the DataSet
+    name = function(value) {
+      if (missing(value)) {
+        return(self$dataRepository$name)
+      }
+      self$dataRepository$name <- value
+    },
+    #' @field xDimension Dimension in which the xValues are defined
+    xDimension = function(value) {
+      if (missing(value)) {
+        return(private$.xValues$dimension)
+      }
+      private$setColumnDimension(private$.xValues, value)
+    },
+    #' @field xUnit Unit in which the xValues are defined
+    xUnit = function(value) {
+      if (missing(value)) {
+        return(private$.xValues$displayUnit)
+      }
+      private$setColumnUnit(private$.xValues, value)
+    },
+    #' @field xValues Values stored in the xUnit dimension (not necessarily in the base unit of the dimension)
+    xValues = function(values) {
+      if (missing(values)) {
+        return(private$getColumnValues(private$.xValues))
+      }
+
+      private$setColumnValues(private$.xValues, values)
+    },
+    #' @field yDimension Dimension in which the xValues are defined
+    yDimension = function(value) {
+      if (missing(value)) {
+        return(private$.yValues$dimension)
+      }
+      private$setColumnDimension(private$.yValues, value)
+    },
+    #' @field yUnit Unit in which the yValues are defined
+    yUnit = function(value) {
+      if (missing(value)) {
+        return(private$.yValues$displayUnit)
+      }
+      private$.yValues$displayUnit <- value
+    },
+    #' @field yValues Values stored in the yUnit dimension (not necessarily in the base unit of the dimension)
+    yValues = function(values) {
+      if (missing(values)) {
+        return(private$getColumnValues(private$.yValues))
+      }
+
+      private$setColumnValues(private$.yValues, values)
+    }
+  ),
+  public = list(
+    dataRepository = NULL,
+    #' @description
+    #' Initialize a new instance of the class
+    #' @param dataRepository Instance of the \code{DataRepository} object to wrap
+    #' @return A new `DotNetWrapper` object.
+    initialize = function(dataRepository = NULL) {
+      self$dataRepository <- dataRepository %||% private$createDataRepository()
+      private$initializeCache()
+    },
+    #' @description
+    #' Print the object to the console
+    #' @param ... Rest arguments.
+    print = function(...) {

```

Comment:
@hannaei and @PavelBal 
Here the print method should be enhanced to show the data in a user friendly fashion. I'll leave that to you